### PR TITLE
Authorization Policies

### PR DESCRIPTION
**Pull request by lonnieezell**
Implements new authorization policies for more flexible authorization of actions.

### CODE REVIEW

File: app/Libraries/Policies/Policy.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    private array $policies = [];
+    private User $user;
+
+    /**
+     * Used within the controllers to determine if the current
+     * user has the ability to perform the given action.
+     *
+     * Example:
+     * if (! $this->policy->can('posts.create')) {
+     *    return $this->policy->deny(403, 'You are not allowed to create posts.');
```
Comment (datamweb) [inline app/Libraries/Policies/Policy.php]:
```suggestion
     *    return $this->policy->deny('You are not allowed to create posts.', 403);
```

File: _docs/index.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # Forum Developer Guide
 
 [Getting Started](./start.md)
+[Autnorization Policies](./policies.md)
```
Comment (datamweb) [inline _docs/index.md]:
```suggestion
- [Getting Started](./start.md)
- [Autnorization Policies](./policies.md)
```

File: _docs/policies.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+For example, a user may be allowed to edit a post, but only if they are the author of the post. This is not possible with Shield alone, but can be achieved with a policy that checks the user's ID against the post's author ID.
+
+## Creating Policies
+
+Policies are simple classes that extend the `App\Libraries\Policies\PolicyInterface` class. These work hand-in-hand with the permissions defined in `App\AuthGroups` and the names of the classes and methods are based on these permissions. For example, if you have a permission called `posts.create`, you would create a policy called `PostPolicy` with a method called `create()`. The name must be the singular version of the first segment of the permission. In the previous example, the class name is `PostPolicy` even though the permission's first segment is `posts`. The file is expected to be located within the `app/Policies` directory and have the appropriate namespace, `App\Policies\PostPolicy`.
```
Comment (datamweb) [inline _docs/policies.md]:
```suggestion
Policies are simple classes that extend the `App\Libraries\Policies\PolicyInterface` class. These work hand-in-hand with the permissions defined in `App\AuthGroups` and the names of the classes and methods are based on these permissions. For example, if you have a permission called `posts.create`, you would create a policy called `PostPolicy` with a method called `create()`. The name must be the singular version of the first segment of the permission. In the previous example, the class name is `PostPolicy` even though the permission's first segment is `posts`. The file is expected to be located within the `app/Policies` directory and have the appropriate namespace, `App\Policies`.
```

**Review by michalsn — APPROVED** [review]
Looks great!

Failed checks can be handled later - even after merge.

**Comment by lonnieezell** [comment]
> Looks great!
> 
> Failed checks can be handled later - even after merge.

Alrighty then! I'll go ahead and merge.

**Pull request merged**